From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Greg Kroah-Hartman <gregkh@google.com>
Date: Tue, 13 Oct 2020 18:22:27 +0200
Subject: ANDROID: disable ANDROID_VENDOR_HOOKS

Commit d25e37d89dd2 ("tracepoint: Optimize using static_call()") changed
the format of DEFINE_TRACE() and other tracepoint logic, which broke the
ANDROID_VENDOR_HOOKS logic.

As I can't figure out how to unwind the build traces for the errors,
disable this for now so the merge can continue.

Bug: 156285741
Cc: Todd Kjos <tkjos@google.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@google.com>
Change-Id: I6794bdec76ba25370891747b296ea30a51812982
---
 arch/arm64/configs/gki_defconfig | 1 -
 arch/x86/configs/gki_defconfig   | 1 -
 drivers/android/Kconfig          | 1 +
 3 files changed, 1 insertion(+), 2 deletions(-)

diff --git a/arch/arm64/configs/gki_defconfig b/arch/arm64/configs/gki_defconfig
--- a/arch/arm64/configs/gki_defconfig
+++ b/arch/arm64/configs/gki_defconfig
@@ -466,7 +466,6 @@ CONFIG_RAS=y
 CONFIG_ANDROID=y
 CONFIG_ANDROID_BINDER_IPC=y
 CONFIG_ANDROID_BINDERFS=y
-CONFIG_ANDROID_VENDOR_HOOKS=y
 CONFIG_LIBNVDIMM=y
 # CONFIG_ND_BLK is not set
 CONFIG_INTERCONNECT=y
diff --git a/arch/x86/configs/gki_defconfig b/arch/x86/configs/gki_defconfig
--- a/arch/x86/configs/gki_defconfig
+++ b/arch/x86/configs/gki_defconfig
@@ -404,7 +404,6 @@ CONFIG_RAS=y
 CONFIG_ANDROID=y
 CONFIG_ANDROID_BINDER_IPC=y
 CONFIG_ANDROID_BINDERFS=y
-CONFIG_ANDROID_VENDOR_HOOKS=y
 CONFIG_LIBNVDIMM=y
 # CONFIG_ND_BLK is not set
 CONFIG_INTERCONNECT=y
diff --git a/drivers/android/Kconfig b/drivers/android/Kconfig
--- a/drivers/android/Kconfig
+++ b/drivers/android/Kconfig
@@ -57,6 +57,7 @@ config ANDROID_BINDER_IPC_SELFTEST
 config ANDROID_VENDOR_HOOKS
 	bool "Android Vendor Hooks"
 	depends on TRACEPOINTS
+	depends on BROKEN
 	help
 	  Enable vendor hooks implemented as tracepoints
 
